src/Microsoft.AspNetCore.Routing/DefaultLinkGenerator.cs

```diff
@@ -41,13 +41,22 @@ public string GetLink(LinkGeneratorContext context)
 
         public bool TryGetLink(LinkGeneratorContext context, out string link)
         {
-            var address = context.Address;
-            var endpoints = _endpointFinder.FindEndpoints(address);
+            IEnumerable<Endpoint> endpoints = null;
             link = null;
 
-            if (endpoints == null)
+            // If endpoints were already provided by another, use them, else find the endpoints using address's name
+            if (context.Endpoints == null)
             {
-                return false;
+                var address = context.Address;
+                endpoints = _endpointFinder.FindEndpoints(address);
+                if (endpoints == null)
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                endpoints = context.Endpoints;
```

Comment:
This really feels like it's two different APIs or abstractions rather than a single method